### DISCUSSION TITLE
[Azure Key Vault] Multi vault integration for same project/namespace

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#406] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

v12.0.2

### Description

Hi guys,
a client uses Azure Key Vault and we are working to integrate it with the Console.

The client has the following vaults setup:
* _N_ generic vault, with secrets that can be shared through different projects/areas (e.g. `main-database-dev-vault`, `main-database-prod-vault`, `shared-redis-dev-vault` and so on);
* 1 specific vault for <project, environment> (e.g. `project1-dev-vault`, `project1-prod-vault` and so on).

Is the console able to handle this setup?

For example, a configuration like the following one on the "**Speed check**" project:
* **development** env &rarr; vaults `network-mysql-development`, `network-gmaps-development` and `speed-check-development`;
* **preproduction** env &rarr; vaults `network-mysql-qa`, `network-gmaps-qa` and `speed-check-qa`;
* **production** env &rarr; vaults `network-mysql-prod`, `network-gmaps-prod` and `speed-check-prod`;

With the `network-mysql-{{ENV}}` and `network-gmaps-{{ENV}}` vaults that are shared between multiple projects and the `speed-check-{{ENV}}` vaults that are specific for the project.

Thanks.

**Comment by mattia-fumo** [comment]
Hi,
At the moment the Console supports just one store per environment

**Reply by frengooo** [reply]
Do you think that extend the support to multiple stores have a big effort?
Or is there a workaround to decorate the External Secret Operator manifests to get secrets of other vaults?

**Reply by mattia-fumo** [reply, marked as answer]
> Do you think that extend the support to multiple stores have a big effort?

It would require extensive analysis of the desired architecture. At a first look i'd say it's a big deal.

> Or is there a workaround to decorate the External Secret Operator manifests to get secrets of other vaults?

Nothing that I'm aware of.

**Reply by frengooo** [reply]
So, to integrate with different vaults, we should:
- manually create the generic vaults manifests (maybe in the project template);
- let the Console create the project-specific vault manifest.

Thanks.
